File: lua/typing_kata/ui/stats.lua
-- Stats display screen
local M = {}

local buffer_utils = require('typing_kata.ui.buffer')
local ranks = require('typing_kata.core.ranks')
local config = require('typing_kata.config')

-- Helper to center text
local function center(text, width)
  local padding = math.max(0, math.floor((width - vim.fn.strdisplaywidth(text)) / 2))
  return string.rep(' ', padding) .. text
end

-- Format time
local function format_time(seconds)
  local hours = math.floor(seconds / 3600)
  local mins = math.floor((seconds % 3600) / 60)
  if hours > 0 then return string.format('%dh %dm', hours, mins) else return string.format('%dm', mins) end
end

function M.show(player)
  local buf = buffer_utils.create_scratch_buffer('Typing Kata Stats')
  M.render_stats(buf, player)

  local ui_config = config.current.ui
  local win = buffer_utils.create_floating_window(buf, {
    width = 70,
    height = 30,
    position = ui_config.menu_position,
    border = 'rounded',
    title = ' PLAYER STATISTICS ',
  })

  if not win then return end
  M.setup_keymaps(buf, win, player)
end

function M.render_stats(buf, player)
  local rank = ranks.get_rank_by_xp(player.current_xp)
  local progress = ranks.progress_to_next_rank(player.current_xp)
  local lines = {}
  local width = 66 -- buffer width minus padding roughly

  -- Header
  table.insert(lines, '')
  table.insert(lines, center(string.format('PLAYER: %s', player.name:upper()), width))
  table.insert(lines, center(string.format('%s %s', rank.symbol, rank.name), width))
  table.insert(lines, '')

  -- Global Stats Grid
  table.insert(lines, '╭─ GLOBAL STATS ─────────────────────────────────────────────────╮')
  table.insert(lines, string.format('│  XP: %-12d  Playtime: %-12s  Sessions: %-4d   │', 
    player.current_xp, format_time(player.total_playtime), player.total_sessions))
  table.insert(lines, '╰────────────────────────────────────────────────────────────────╯')
  table.insert(lines, '')

  -- Modes Grid
  table.insert(lines, '  MODE PERFORMANCE')
  table.insert(lines, '  ────────────────')

  local mode_names = {
    { key = 'word_typing', name = 'Word Typing' },
    { key = 'snake_apple', name = 'Snake Apple' },
    { key = 'vim_motions', name = 'Vim Motions' },
    { key = 'coding_lessons', name = 'Coding Lessons' },
    { key = 'symbol_training', name = 'Symbols' },
    { key = 'nvim_command_quiz', name = 'Command Quiz' },
  }

  for _, mode in ipairs(mode_names) do
    local s = player.stats[mode.key]
    if s and s.tasks_completed > 0 then
      local title = string.format('  %s %s', '◆', mode.name)
      table.insert(lines, title)
      local stat_line = string.format('    Accuracy: %-5.1f%%   Streak: %-4d   XP: %-6d', 
        s.total_accuracy, s.best_streak, s.total_xp_earned)
      table.insert(lines, stat_line)
      table.insert(lines, '')
    end
  end

  -- Empty state if no games played
  if player.total_sessions == 0 then
    table.insert(lines, '')
    table.insert(lines, center('No games played yet.', width))
    table.insert(lines, center('Go play some games!', width))
  end

  table.insert(lines, '')
  table.insert(lines, center('[ESC] Back', width))

  buffer_utils.set_content(buf, lines)
end

function M.setup_keymaps(buf, win, player)
  local opts = { buffer = buf, noremap = true, silent = true }
  local function close()
    if vim.api.nvim_win_is_valid(win) then vim.api.nvim_win_close(win, true) end
    require('typing_kata.ui.menu').show(player)
  end
  vim.keymap.set('n', 'q', close, opts)
  vim.keymap.set('n', '<Esc>', close, opts)
end

return M

File: python_archive/src/ui/screens/stats.py
"""Stats and progress screen."""
from textual.app import ComposeResult
from textual.screen import Screen
from textual.widgets import Header, Footer, Button, Static
from textual.containers import Container, Vertical, ScrollableContainer

from ...utils.stats_calculator import format_time


class StatsScreen(Screen):
    """Display player stats and progress."""

    CSS = """
    StatsScreen {
        align: center top;
    }

    #stats-container {
        width: 80;
        height: auto;
        padding: 2;
        margin: 1;
    }

    .stats-section {
        padding: 1;
        margin: 1;
        border: solid $primary;
        background: $surface;
    }

    Button {
        width: 30;
        margin: 1;
    }
    """

    def compose(self) -> ComposeResult:
        """Compose the stats screen."""
        yield Header()

        yield ScrollableContainer(
            Static("[bold cyan]Player Statistics & Progress[/]\n", id="title"),
            Static(self._get_overall_stats(), classes="stats-section"),
            Static(self._get_mode_stats(), classes="stats-section"),
            Button("← Back to Menu", id="back", variant="primary"),
            id="stats-container"
        )

        yield Footer()

    def _get_overall_stats(self) -> str:
        """Get overall player statistics."""
        player = self.app.player
        rank = self.app.rank_system.get_rank(player.current_rank)

        lines = []
        lines.append("[bold yellow]Overall Progress[/]")
        lines.append("")
        lines.append(f"[cyan]Player Name:[/] {player.name}")
        if rank:
            lines.append(f"[cyan]Current Rank:[/] {rank.symbol} {rank.name} ({player.current_rank + 1}/100)")
        lines.append(f"[cyan]Total XP:[/] {player.current_xp:,}")
        lines.append(f"[cyan]Total Sessions:[/] {player.total_sessions}")
        lines.append(f"[cyan]Total Playtime:[/] {format_time(player.total_playtime)}")
        lines.append(f"[cyan]Account Created:[/] {player.created_at[:10]}")
        lines.append(f"[cyan]Last Played:[/] {player.last_played[:10]}")

        return "\n".join(lines)

    def _get_mode_stats(self) -> str:
        """Get per-mode statistics."""
        player = self.app.player

        if not player.stats:
            return "[dim]No game mode statistics yet. Start playing to track your progress![/]"

        mode_names = {
            'snake_apple': '🐍 Snake Apple',
            'symbol_training': '🔣 Symbol Training',
            'coding_lessons': '💻 Coding Lessons',
            'word_training': '📝 Word Training',
            'vim_motions': '⚡ Vim Motions',
            'comprehensive_keys': '⌨️ Comprehensive Keys',
        }

        lines = []
        lines.append("[bold yellow]Game Mode Statistics[/]")
        lines.append("")

        for mode_key, stats in player.stats.items():
            mode_name = mode_names.get(mode_key, mode_key.replace('_', ' ').title())
            lines.append(f"[bold cyan]{mode_name}[/]")
            lines.append(f"  Tasks Completed: {stats.tasks_completed}")
            lines.append(f"  Average Accuracy: {stats.total_accuracy:.1f}%")
            lines.append(f"  Average Speed: {stats.average_speed:.2f}")
            lines.append(f"  Best Streak: {stats.best_streak}")
            lines.append(f"  Time Played: {format_time(stats.total_time_played)}")
            lines.append(f"  XP Earned: {stats.total_xp_earned:,}")
            lines.append("")

        return "\n".join(lines)

    def on_button_pressed(self, event: Button.Pressed) -> None:
        """Handle button presses."""
        if event.button.id == "back":
            self.app.pop_screen()
